hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/StorageBasedLockProvider.java

```diff
@@ -590,4 +612,45 @@ private void logErrorLockState(LockState state, String msg) {
   long getCurrentEpochMs() {
     return System.currentTimeMillis();
   }
+
+  /**
+   * Helper method to start audit operation when lock is acquired.
+   */
+  private void startAuditOperation(long timestamp) {
+    storageLpAuditService.ifPresent(auditService -> {
+      try {
+        String sessionId = UUID.randomUUID().toString();
+        auditService.startOperation(sessionId, timestamp);
+      } catch (Exception e) {
+        // Log but don't fail the lock operation due to recording failures
+        logger.warn("Owner {}: Failed to start audit operation: {}", ownerId, e.getMessage());
+      }
+    });
+  }
+
+  /**
+   * Helper method to update audit operation for renewals.
+   */
+  private void updateAuditOperation(long timestamp) {
+    storageLpAuditService.ifPresent(auditService -> {
+      try {
+        auditService.updateOperation(timestamp);
+      } catch (Exception e) {
+        logger.warn("Owner {}: Failed to update audit operation: {}", ownerId, e.getMessage());
+      }
+    });
+  }
+
+  /**
+   * Helper method to end audit operation.
+   */
+  private void endAuditOperation(long timestamp, String finalState) {
```

Comment:
wondering if we should just have enums for well-defined states, and then only a single API that has the enum state (e.g., RENEW) with the timestamp?